### PR TITLE
Adding Pod Disruption Budget (PDB) Configuration to Strimzi Kafka Operator's Helm Chart

### PR DESCRIPTION
**Pull request by haijun2022**
Adding Pod Disruption Budget (PDB) Configuration to Strimzi Kafka Operator's Helm Chart

### Type of change
- Enhancement / new feature

### Description
refer to issue - https://github.com/strimzi/strimzi-kafka-operator/issues/8607

Related problem
The Strimzi Kafka Operator supports high availability based on the kube lease method from version 0.32 onwards. This is a valuable feature, particularly given the current reliance on SPS objects.

However, it seems there's currently no way to configure Pod Disruption Budgets (PDBs) within the operator's Helm chart. PDBs are important for maintaining a minimum number of operational pods during disruptions.

Solution
This PR enhancing the Strimzi Kafka Operator's Helm chart by introduced an option to configure PDBs including 

- PDB configuration file
- Values file for PDB configuration

This feature would improve the resilience of our deployments, especially in disruption scenarios by ensuring a minimum number of pods remain operational.

### Checklist
- [x] Make sure all tests pass

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Thanks for the PR. But you have to make the change to `packaging/helm-chart`. The `helm-chart/` diretory can change only as a part of a Strimzi release.

I wonder if you should name it `051-...` rather then `061-...`. Also, can `maxUnavailable` be used in parallel with `minAvailable`? You should also add it to the list of options in the README.md file.

**Comment by haijun2022** [comment]
thank you, I will review and update the pr accordingly! @scholzj

**Comment by scholzj** [comment]
@haijun2022 Any plans to get back to this? Thanks.

**Comment by haijun2022** [comment]
Hi @scholzj, I am sorry for the slow response because we have company internal code review processes for such open-source contribution requests(including any further changes in same PR.  Our company continually improving the process but still need more time.

I understand it already taken such longer time, I am ok if  this is urgent for planed release and others can work on it!

**Comment by scholzj** [comment]
@haijun2022 Ok, thanks for the update. It is not that we need this feature urgently. We are just trying to avoid stale PRs being opened.

File: helm-charts/helm3/strimzi-kafka-operator/values.yaml [hunk 1]
```diff
@@ -167,4 +167,4 @@ labelsExclusionPattern: ""
 # Controls whether Strimzi generates network policy resources (By default true)
 generateNetworkPolicy: true
 # Override the value for Connect build timeout
-connectBuildTimeoutMs: 300000
+connectBuildTimeoutMs: 300000
```
Comment (scholzj) [inline helm-charts/helm3/strimzi-kafka-operator/values.yaml:170]: The changes look better now. But you would need to revert this new-line change to get the build to pass.

File: helm-charts/helm3/strimzi-kafka-operator/values.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ watchAnyNamespace: false
 
 defaultImageRegistry: quay.io
 defaultImageRepository: strimzi
-defaultImageTag: 0.37.0
+defaultImageTag: 0.38.0
```
Comment (scholzj) [inline helm-charts/helm3/strimzi-kafka-operator/values.yaml:13]: The point is, that this file should not be changed at all. You can change only the files in `./packaging/`.

Comment (haijun2022) [reply helm-charts/helm3/strimzi-kafka-operator/values.yaml]: thanks a lot @scholzj. I was a little bit confused and I am clear now. Just updated the PR and please review. 👍

**Review by scholzj — APPROVED** [review]
Looks good. Thanks a lot for the PR.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks for the PR!

**Comment by haijun2022** [comment]
Thanks a lot @scholzj  and @ppatierno for your review and kind suggestions!

**Pull request merged**